test.sh

```diff
@@ -1,17 +1,22 @@
 #!/usr/bin/env bash
 
 fails=0
-for i in $(grep -o 'travis.*\.svg' README.md); do
-  url=$(echo $i | sed "s/\//\/repositories\//" | sed "s/svg/json/")
-  curl -sq https://api.$url | grep 'last_build_result":0' > /dev/null
+total=0
+while read -r -a plugin; do
+  total=$((total+1))
+  travis_web_url="https://${plugin[0]}/${plugin[1]}"
+  url="https://api.${plugin[0]}/repos/${plugin[1]}/branches/${plugin[2]}"
+  curl -sq -H 'Accept: application/vnd.travis-ci.2.1+json' ${url} | grep '"state":"passed"' &> /dev/null
```

Comment:
Instead of `grep '...' &> /dev/null`, then capturing `$?`, you could `grep --quiet '...'` and capture its status directly.

L9-L11 could be reduced to something like:

```
if (curl -sq -H 'Accept: application/vnd.travis-ci.2.1+json' ${url} | grep --quiet '"state":"passed"'); then
  # Build passed.
else
  # Build failed.
fi
```

Reply:
Yah, was kind of a holdout of testing and the old version : ) Updated.